src/logging.ts

```diff
@@ -0,0 +1,99 @@
+import db from "./db.ts";
+import { Config } from "./schemas/config.ts";
+import { LoggingWhitelist } from "./schemas/loggingWhitelist.ts";
+import type { TextChannel, Guild, Role } from "discord.js";
+import { eq, and, sql } from "drizzle-orm";
+const { placeholder } = sql;
+
+export enum LogMode {
+	NONE = 0,
+	DELETES = 1 << 0,
+	EDITS = 1 << 1,
+	"DELETES & EDITS" = DELETES | EDITS,
+}
+
+export type LogConfig =
+	| {
+			mode: LogMode.NONE;
+			channel: null;
+	  }
+	| {
+			mode: LogMode;
+			channel: TextChannel["id"];
+	  };
+
+const logModes = new Map<Guild["id"], LogConfig>();
```

Comment:
Why are you caching the log mode? Reading it from SQLite directly should be cheap enough, even for common events.

Reply:
Honestly I'm not sure myself, I'll remove that